### PR TITLE
Orca: fix tf estimator ut random fail

### PR DESCRIPTION
**Pull request by hamham223**
### 1. Why the change?

Fix https://github.com/analytics-zoo/arda-docker/issues/729

Optimizer: SGD -> Adam

```python
            for i in range(len(pre_opt_weights)):
                assert np.array_equal(pre_opt_weights[i], after_opt_weights[i])
```

### 2. How to test?
- [x] Orca-Pytest-Ray-Py37-Spark3
- [x] Orca-Pytest-Ray-Py38-Spark3

### CODE REVIEW

**Comment by hamham223** [comment]
@sgwhat Let's run the test again to see whether the random failure is eliminated?

**Comment by hamham223** [comment]
Hooray! no piles of `nan` anymore

**Comment by hamham223** [comment]
@hkvision take a look? Now `Orca-Pytest-Ray-Py37/38-Spark3` are the longest two tests, we might need to split them into two parts?

**Comment by hkvision** [comment]
> @hkvision take a look? Now `Orca-Pytest-Ray-Py37/38-Spark3` are the longest two tests, we might need to split them into two parts?

One thing we can do is to put mxnet related tests into horovod?

File: python/orca/test/bigdl/orca/learn/spark/test_tf2_estimator.py [hunk 1]
```diff
@@ -41,7 +41,7 @@ def compile_args(config):
     else:
         lr = 1e-3
     args = {
-        "optimizer": tf.keras.optimizers.SGD(lr),
+        "optimizer": tf.keras.optimizers.Adam(lr),
```
Comment (hkvision) [inline python/orca/test/bigdl/orca/learn/spark/test_tf2_estimator.py:44]: SGD will possibly get gradient disappear and get nan weights; thus changed to adam
